internal/provider/vaultsecrets/data_source_vault_secrets_rotating_secret.go

```diff
@@ -0,0 +1,148 @@
+				Required:    true,
+			},
+			"secret_name": schema.StringAttribute{
+				Description: "The name of the Vault Secrets secret.",
+				Required:    true,
+			},
+			"secret_values": schema.MapAttribute{
+				Description: "The secret values corresponding to the secret name input.",
+				Computed:    true,
+				Sensitive:   true,
+				ElementType: types.StringType,
+			},
+			"secret_version": schema.Int64Attribute{
+				Description: "The version of the Vault Secrets secret.",
+				Computed:    true,
+			},
+			"secret_provider": schema.StringAttribute{
+				Description: "The name of the provider this rotating secret is for",
+				Computed:    true,
+			},
+			"organization_id": schema.StringAttribute{
+				Description: "The ID of the HCP organization where the Vault Secrets app is located.",
+				Computed:    true,
+			},
+			"project_id": schema.StringAttribute{
+				Description: "The ID of the HCP project where the Vault Secrets app is located.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func (d *DataSourceVaultSecretsRotatingSecret) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*clients.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *clients.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+	d.client = client
+}
+
+func (d *DataSourceVaultSecretsRotatingSecret) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data DataSourceVaultSecretsRotatingSecretModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	client := d.client
+	if client == nil {
+		resp.Diagnostics.AddError(
+			"Unconfigured HCP Client",
+			"Expected configured HCP client. Please report this issue to the provider developers.",
+		)
+		return
+	}
+
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+
+	openSecret, err := clients.OpenVaultSecretsAppSecret(ctx, client, loc, data.AppName.ValueString(), data.SecretName.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(err.Error(), "Unable to open secret")
+		return
+	}
+
+	var secretValues map[string]string
+	var secretVersion int64
+	switch {
+	case openSecret.RotatingVersion != nil:
+		secretValues = openSecret.RotatingVersion.Values
+		secretVersion = openSecret.RotatingVersion.Version
+	default:
+		resp.Diagnostics.AddError(
+			"Unsupported HCP Secret type",
+			fmt.Sprintf("HCP Secrets secret type %q is not currently supported by terraform-provider-hcp", openSecret.Type),
+		)
+		return
+	}
+
+	secretsOutput, diag := types.MapValueFrom(ctx, types.StringType, secretValues)
+	resp.Diagnostics.Append(diag...)
+
+	// TODO: what is ID supposed to be?
+	// data.ID = ?
```

Comment:
Is this ID actually used for unique identification of data sources or by anything TF-provider-specific? If not, maybe one option is to just remove it altogether.

Reply:
im honestly not sure, would like some feedback from @hashicorp/cloud-foundations 
otherwise maybe something like
`project/<project id>/app/<app name>/secret/<secret name>`
wdyt?

Reply:
If our goal here is a unique resource identifier, this will almost work. The only scenario where I see it failing is if you delete the secret and re-create it will the same name. However, this is much better than the AppName that we are using elsewhere.

My preferred solution would still be to remove it if possible.